packages/nimble-components/src/table-column/icon/group-header-view/index.ts

```diff
@@ -0,0 +1,77 @@
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { ViewTemplate, observable } from '@microsoft/fast-element';
+import { styles } from './styles';
+import { template } from './template';
+import type { TableColumnEnumColumnConfig } from '../../enum-base';
+import type { TableFieldValue } from '../../../table/types';
+import { TableColumnTextGroupHeaderViewBase } from '../../text-base/group-header-view';
+import type { IconSeverity } from '../../../icon-base/types';
+import {
+    MappingIconConfig,
+    type IconView
+} from '../../enum-base/models/mapping-icon-config';
+import { MappingSpinnerConfig } from '../../enum-base/models/mapping-spinner-config';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-icon-group-header-view': TableColumnIconGroupHeaderView;
+    }
+}
+
+/**
+ * The group header view for the icon column
+ */
+export class TableColumnIconGroupHeaderView
+    extends TableColumnTextGroupHeaderViewBase<
+    TableFieldValue,
+    TableColumnEnumColumnConfig
+    >
+    implements IconView {
+    @observable
+    public severity: IconSeverity;
+
+    @observable
+    public iconTemplate?: ViewTemplate<IconView>;
+
+    @observable
+    public visual?: 'spinner' | 'icon';
+
+    private columnConfigChanged(): void {
+        this.updateState();
+    }
+
+    private groupHeaderValueChanged(): void {
+        this.updateState();
+    }
+
+    private updateState(): void {
+        this.visual = undefined;
+        if (!this.columnConfig?.mappingConfigs) {
```

Comment:
mappingConfigs isn;t nullable right?


```suggestion
        if (!this.columnConfig) {
```